core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerService.java

```diff
@@ -22,18 +22,25 @@
 import java.util.stream.Collectors;
 
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 
+import org.springframework.context.event.ApplicationEventMulticaster;
 import org.springframework.stereotype.Component;
 
+import com.expediagroup.streamplatform.streamregistry.core.events.EventType;
+import com.expediagroup.streamplatform.streamregistry.core.events.NotificationEvent;
+import com.expediagroup.streamplatform.streamregistry.core.events.NotificationEventEmitter;
 import com.expediagroup.streamplatform.streamregistry.core.handlers.HandlerService;
 import com.expediagroup.streamplatform.streamregistry.core.repositories.ConsumerRepository;
 import com.expediagroup.streamplatform.streamregistry.core.validators.ConsumerValidator;
 import com.expediagroup.streamplatform.streamregistry.model.Consumer;
 import com.expediagroup.streamplatform.streamregistry.model.keys.ConsumerKey;
 
+@Slf4j
 @Component
 @RequiredArgsConstructor
-public class ConsumerService {
+public class ConsumerService implements NotificationEventEmitter<Consumer> {
```

Comment:
We've discussed already but to capture the comment here - would make for less repetition if the `NotificationEventEmitter` were composed into the Service rather than inheriting it.

Reply:
Ah didn't see comment above. I'll resolve this thread.